### PR TITLE
add swap support

### LINKED ISSUE #93: Swap support
In addition to `memory()`-usage also report swap usage. I would be willing to implement this myself.

Would you prefer this as a separate `swap()` function or as a part of `memory()`? Some platform-APIs will return memory and swap information together.

### PR DESCRIPTION
**Pull request by davidkna**
This PR adds `swap` to `Platform` and implements the required functionality for Linux, macOS and Windows. Further, `TryFrom` is implemented between `Memory` and `Swap` based on per-platform compatibility (Linux and Windows are supported).

Closes #93

### CODE REVIEW

File: examples/info.rs [hunk 1]
```diff
@@ -75,6 +75,11 @@ fn main() {
         Err(x) => println!("\nMemory: error: {}", x)
     }
 
+    match sys.swap() {
+        Ok(swap) => println!("\nSwap: {} used / {} ({} bytes) total ({:?})", saturating_sub_bytes(swap.total, swap.free), swap.total, swap.total.as_u64(), swap.platform_swap),
+        Err(x) => println!("\nMemory: error: {}", x)
```
Comment (unrelentingtech) [inline examples/info.rs:80]: This line should also say 'Swap:'

**Comment by unrelentingtech** [comment]
hmm. I don't think the `TryFrom` thing is a very good API. The simpler, more obvious way to leverage the fact that some platforms provide both in one syscall would be to just have all the retrieval options as separate functions:

- `fn memory(&self) -> io::Result<Memory>`
- `fn swap(&self) -> io::Result<Swap>`
- `fn memory_and_swap(&self) -> io::Result<(Memory, Swap)> /// On some platforms this is more efficient than calling memory() and swap() separately`

**Comment by davidkna** [comment]
Done!

**Comment by unrelentingtech** [comment]
Thanks!

**Pull request merged**